convert2rhel/unit_tests/utils_test.py

```diff
@@ -874,7 +888,312 @@ def communicate(self):
         pass
 
     def poll(self):
-        return 0
+        return self.return_code
+
+    def close(self):
+        pass
+
+
+@pytest.mark.parametrize(
+    ("output", "skip_stdout_decode", "expected"),
+    (
+        (DummyPopenOutput([u"test".encode()]), True, b"test"),
+        (DummyPopenOutput([u"testê café".encode("utf-8")]), False, u"testê café"),
+        (DummyPopenOutput([u"testê café".encode("utf-8")]), False, u"testê café"),
+        (DummyPopenOutput([u"test".encode()]), False, "test"),
+    ),
+)
+def test_normalize_subprocess_output(output, skip_stdout_decode, expected):
+    result = utils._normalize_subprocess_output(output.stdout.readline, skip_stdout_decode)
+    assert result == expected
```

Comment:
```suggestion
    assert result == expected
    assert isinstance(result, type(expected))
```

On python2, ascii-only byte strings and ascii-only unicode strings that contain the same abstract characters are equal.  We actually care about this since we don't want to ensure that this function always returns bytes when skip_stdout_decode is True and return text when skip_stdout_decode is False.  Adding the isinstance assert will catch this cornercase.

Reply:
Not resolving this comment, but we have this mapped under the gist mentioned in a comment here, so should be fine to resolve it anyway.